### ISSUE TITLE
New improvement - group elements by

### ISSUE THREAD

**Issue by h1560506** [w11k/w11k-select#28]
Hi there,

I would like to thank you for the awesome plugin.
Everything works great, but unfortunately I miss one thing which is group elements by.
I mean it should be similar like this : 

`<ul ng-repeat="(key, value) in options | orderBy: 'group' | groupBy:group">
      <li>{{key}}
        <ul>
          <li ng-repeat="option in value | orderBy: 'group2'">
            {{option.label}}
          </li>
        </ul>
      </li>
    </ul>`

It is possible/Are you going to implement group functionality to this plugin or I should build this by myself ?

Best regards,

**Comment by pburgmer** [comment]
Hi,

I tried it some time ago but it was very difficult to find a generic solution, so i aborted.
A customer asked for that feature to, so I will try again in the next few month ;)

Best regards,
Philipp

**Comment by pburgmer** [comment]
Hi,

We implemented a hierarchy feature in version 0.10.
You can configure which property of your data object contains the children.

```
config = {
  "children": "subItems",
  "multiple": true
}
```

and
 
```
data = [
  {
    label: 'Grandfather1a',
    value: '1das',
    subItems: [
      {
        label: 'Father1A',
        value: '2ads',
        subItems: [
          {
            label: 'Child1a',
            value: '21asd32',
            subItems: [
              {
                label: 'GrandChild1a',
                value: '2213123132',
                subItems: []
              },
              {
                label: 'GrandChild2a',
                value: '3asddasdas',
                subItems: []
              }
            ]
          },
          {
            label: 'Child2a',
            value: '3asdas',
            subItems: []
          }
        ]
      },
      {
        label: 'Father2a',
        value: '3asd213e',
        subItems: [
          {
            label: 'Child1a',
            value: '12312das2',
            subItems: []
          },
          {
            label: 'Child2a',
            value: '3asda14',
            subItems: []
          }
        ]
      }
    ]
  }
]
```

**Comment by pburgmer** [comment]
We added a demo to http://w11k.github.io/w11k-select/
